pkg/kubernetes/upgrade.go

```diff
@@ -0,0 +1,123 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package kubernetes
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/dapr/cli/pkg/print"
+	"github.com/dapr/cli/utils"
+	helm "helm.sh/helm/v3/pkg/action"
+	"k8s.io/helm/pkg/strvals"
+)
+
+var crds = []string{
+	"components",
+	"configuration",
+	"subscription",
+}
+
+type UpgradeConfig struct {
+	RuntimeVersion string
+}
+
+func Upgrade(conf UpgradeConfig) error {
+	helmConf, err := helmConfig("dapr-system")
+	if err != nil {
+		return err
+	}
+
+	daprChart, err := daprChart(conf.RuntimeVersion, helmConf)
+	if err != nil {
+		return err
+	}
+
+	sc, err := NewStatusClient()
+	if err != nil {
+		return err
+	}
+
+	status, err := sc.Status()
+	if err != nil {
+		return err
+	}
+
+	if len(status) == 0 {
+		return errors.New("dapr is not installed in your cluster")
+	}
+
+	print.InfoStatusEvent(os.Stdout, "Dapr control plane version %s detected", status[0].Version)
+
+	upgradeClient := helm.NewUpgrade(helmConf)
+	upgradeClient.ResetValues = true
+	upgradeClient.Namespace = status[0].Namespace
+	upgradeClient.CleanupOnFail = true
+	upgradeClient.Wait = true
+
+	print.InfoStatusEvent(os.Stdout, "Starting upgrade...")
+
+	mtls, err := IsMTLSEnabled()
```

Comment:
should using old certs during upgrade be a flag? `--mtls` Only export certs during upgrade if that flag is provided.

Reply:
Its our recommended best practice to carry the certs over. In this PR, if mTLS is used, they are carried over.